config/post-install/cleanup.yaml

```diff
@@ -0,0 +1,79 @@
+# Copyright 2024 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: batch/v1
+kind: Job
+metadata:
+  generateName: cleanup-serving-
+  namespace: knative-serving
+  labels:
+    app: cleanup-serving
+    app.kubernetes.io/name: knative-serving
+    app.kubernetes.io/component: cleanup-job
+    app.kubernetes.io/version: devel
+spec:
+  ttlSecondsAfterFinished: 600
+  backoffLimit: 10
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "false"
+      labels:
+        app: cleanup-migration-serving
+        app.kubernetes.io/name: knative-serving
+        app.kubernetes.io/component: cleanup-job
+        app.kubernetes.io/version: devel
+    spec:
+      serviceAccountName: controller
+      restartPolicy: OnFailure
+      containers:
+        - name: cleanup
+          # This is the Go import path for the binary that is containerized
+          # and substituted here.
+          image: ko://knative.dev/serving/pkg/cleanup/cmd/cleanup
+          resources:
+            requests:
+              cpu: 100m
+              memory: 100Mi
+            limits:
+              cpu: 1000m
+              memory: 1000Mi
+          securityContext:
+            allowPrivilegeEscalation: false
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            capabilities:
+              drop:
+                - ALL
+            seccompProfile:
+              type: RuntimeDefault
+          env:
+            - name: SYSTEM_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  fieldPath: metadata.namespace
+---
+
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: knative-serving-cleanup
+  labels:
+    serving.knative.dev/controller: "true"
+    app.kubernetes.io/version: devel
+    app.kubernetes.io/name: knative-serving
+rules:
+  - apiGroups: ["rbac.authorization.k8s.io"]
```

Comment:
Should probably limit this to just the single role we need to delete

```yaml
resourceNames: ["..."]
```

Reply:
Oddly this role becomes a lingering resources after this job runs.

I wonder if we should just move this to `serving-core` role?

Reply:
I remove it via the job yes it is a bit odd